### PR TITLE
tools: update CSV generator to output fields for tlv's and subtypes

### PR DESCRIPTION
**Pull request by niftynei**
v1.1 message formats add new types (tlv's and subtypes). this updates
the csv extraction tool to appropriately output them

### CODE REVIEW

File: tools/extract-formats.py [hunk 1]
```diff
@@ -34,42 +50,76 @@ def main(options, args=None, output=sys.stdout, lines=None):
...
-                    match.group('value')), file=output)
+                if tlv:
+                    print("{},{},{}".format(
+                        match.group('name'),
+                        match.group('value'),
+                        tlv), file=output)
+                else:
+                    print("{},{}".format(
+                        match.group('name'),
+                        str(match.group('value') or '$')), file=output)
```
Comment (rustyrussell) [inline tools/extract-formats.py]: I think subtypes should just be a single field, ie. no value.  Rather than a double field with a magic '$' sign?

**Comment by rustyrussell** [comment]
OK, just pushing one unrelated tweak, which fixes parsing of #604 and #593

**Comment by cdecker** [comment]
ACK 33ff4595f96d9ab7574de299669ca2555ed443ff

**Comment by rustyrussell** [comment]
I think this was rolled into #622 ?

**Comment by rustyrussell** [comment]
Closed in favor of #622 which was merged.

**Pull request closed without merging**